Python/instrumentation.c

```diff
@@ -1236,6 +1236,10 @@ _Py_call_instrumentation_jump(
            event == PY_MONITORING_EVENT_BRANCH_RIGHT ||
            event == PY_MONITORING_EVENT_BRANCH_LEFT);
     int to = (int)(dest - _PyFrame_GetBytecode(frame));
+    if (to <= INT_MAX / (int)sizeof(_Py_CODEUNIT)) {
+        PyErr_SetString(PyExc_OverflowError, "instruction offset cannot be converted to a C int");
+        return NULL;
+    }
     PyObject *to_obj = PyLong_FromLong(to * (int)sizeof(_Py_CODEUNIT));
```

Comment:
Sorry, I changed my mind. You should just use a larger type for *to*, and an assertion should be enough if we use `Py_ssize_t`:

```suggestion
    Py_ssize_t to = (dest - _PyFrame_GetBytecode(frame));
    assert(to <= PY_SSIZE_T_MAX / sizeof(_Py_CODEUNIT));
    PyObject *to_obj = PyLong_FromSsize_t(to * sizeof(_Py_CODEUNIT));
```